earthaccess/search.py

```diff
@@ -291,6 +291,23 @@ def debug(self, debug: bool = True) -> Self:
         self._debug = debug
         return self
 
+    def has_granules(self, has_granules: bool = True) -> Self:
+        """Only match collections with granules.
+
+        Parameters:
+            has_granules: If `True`, only return collections with granules. If
+            `False`, only return collections without granules.
+
+        Returns:
+            self
+        """
+        if not isinstance(has_granules, bool):
+            raise TypeError("cloud_hosted must be of type bool")
```

Comment:
```suggestion
            raise TypeError("has_granules must be of type bool")
```

Reply:
Good eye!

Reply:
Good grief! My proofreading skills are in the toilet!